modules/avd/avd.bicep

```diff
@@ -119,3 +132,156 @@ resource workspace 'Microsoft.DesktopVirtualization/workspaces@2019-12-10-previe
 // }
 
 output appGroupName string = appGroupName
+
+
+
+// // Azure Virtual Desktop and Session Hosts region
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-06-01' = [for i in range(0, countAVDInstances): {
+  name: 'vm-redcap-${i}-nic'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: ComputeSubnetId
+          }
+        }
+      }
+    ]
+  }
+}]
+
+resource vm 'Microsoft.Compute/virtualMachines@2023-03-01' = [for i in range(0, countAVDInstances): {
+  name: 'vm-redcap-${i}'
+  location: location
+  identity: AADJoin ? {
+    type: 'SystemAssigned'
+  } : null
+  properties: {
+    licenseType: 'Windows_Client'
+    hardwareProfile: {
+      vmSize: vmSku
+    }
+    osProfile: {
+      computerName: 'vm-redcap-${i}'
+      adminUsername: avdVMAdmin
+      adminPassword: avdVMPassword
+      windowsConfiguration: {
+        enableAutomaticUpdates: false
+        patchSettings: {
+          patchMode: 'Manual'
+        }
+      }
+    }
+    storageProfile: {
+      osDisk: {
+        name: 'vm-OS-${i}'
+        caching: vmDiskCachingType
+        managedDisk: {
+          storageAccountType: vmDiskType
+        }
+        osType: 'Windows'
+        createOption: 'FromImage'
+      }
+      // TODO Turn into params
+      imageReference: {
+        publisher: 'microsoftwindowsdesktop'
+        offer: 'office-365'
+        sku: '20h2-evd-o365pp'
+        version: 'latest'
+      }
+      dataDisks: []
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic[i].id
+        }
+      ]
+    }
+  }
+  dependsOn: [
+    nic[i]
+  ]
+}]
+
+resource domainJoinExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, countAVDInstances): {
+  name: 'AADJoin'
+  parent: vm[i]
+  location: location
+  properties: AADJoin ? {
+    publisher: 'Microsoft.Azure.ActiveDirectory'
+    type: 'AADLoginForWindows'
+    typeHandlerVersion: '1.0'
+    autoUpgradeMinorVersion: true
+    settings: intune ? {
+      mdmId: '0000000a-0000-0000-c000-000000000000' //https://github.com/MicrosoftDocs/azure-docs/issues/94148
+    } : null
+  } : null
+  dependsOn: [
+    vm[i]
+  ]
+}]
+
+resource avdAgentDscExtension 'Microsoft.Compute/virtualMachines/extensions@2023-03-01' = [for i in range(0, countAVDInstances): {
+  name: 'AvdAgentDSC'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Powershell'
+    type: 'DSC'
+    typeHandlerVersion: '2.73'
+    autoUpgradeMinorVersion: true
+    settings: {
+      modulesUrl: artifactsLocation
+      configurationFunction: 'Configuration.ps1\\AddSessionHost'
+      properties: {
+        hostPoolName: hostPool.name
+        registrationInfoToken: hostPool.properties.registrationInfo.token
+        aadJoin: false
+      }
+    }
+  }
+  dependsOn: [
+    vm[i]
```

Comment:
`dependsOn: [ vm[i] ]` is not needed. Should depend on domainJoinExtension (aadJoinExtension).